docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+
+.. code-block:: python
+
+    import nwchemex as nwx 
+    energy = nwx.calculate_scf_energy(molecule = 'H 0. 0. 0. \n H 0. 0. 1.', basis = 'sto-3g')
+
+While the example shows only the two required arguments for the ``nwx.calculate_scf_energy()``
+function, the whole function signature is given below: 
+
+.. code-block:: python
+
+    def calculate_scf_energy(molecule: Union[str, chemist.Molecule], basis:
+    Union[str, simde.type.ao_space], spin: int = 1, max_iterations: int = 50,
+    scf_thresh: float = 1e-10, options: Union[Dict, dataclass] = None, **kwargs) -> float:
+
+In this function, the only required arguments are ``molecule``, which can either
+be a Python ``string`` or a ``chemist.Molecule`` object and ``basis``, which can
+either be a a Python ``string`` or a ``simde.type.ao_space``. Additional SCF
+related options are also explicitly defined in the function with reasonable
+default values. While the number of options are limited for SCF, for correlated
+methods more options may need to be defined on top of the SCF related options.
+One solution to avoid a long list of arguments is to provide additional options
+through ``**kwargs`` argument at the end. However, this would require users to
+know the correct keywords. To help our users to figure out the available
+keywords, we also provide the ``options`` argument, which is an object that
+holds all the possible options with reasonable default values. Our users can
+grab this object and figure out the available options and their default values
+and modify them as they wish. 
+
+Similarly, ``calculate_X_energy`` functions will also be provided for all other
+quantum chemistry methods supported by NWChemEx. Users may also request other
+properties such as the gradients and the wave function using any of the methods. 
+These can be obtained by ``calculate_X_gradients`` and
+``calculate_X_wavefunction``. It should be noted that, caching mechanism of
+NWChemEx avoids any recomputation if the requested quantity is already calculated.
+
+Our UI also provides a more general ``nwx.calculate`` function, where the method
+and the return types can be specified in the ``options`` explicitly or provided
+as key-value arguments in the function call.
+
+.. code-block:: python
+
+    def calculate(molecule: Union[str, chemist.Molecule], method: str, basis:
+    Union[str, simde.type.ao_space], return_energy: Bool = True, options:
+    Union[Dict, dataclass] = None, **kwargs):
+
+Using this function, a user can run any method and request different return
+types either using ``options`` or by setting these arguments explicitly.
+
+.. code-block:: python
+    # Set arguments using options
+    import nwchemex as nwx
+    options = nwx.options(method = 'scf', basis = 'sto-3g', return_gradients=True)
+    energy, gradients = nwx.calculate(molecule = 'H 0. 0. 0. \n H 0. 0. 1.', options = options)
+    
+    # Set arguments explicitly
```

Comment:
Did you mean to start a new code block?

Reply:
Fixed with https://github.com/NWChemEx-Project/.github/commit/c28a7cb41f1d7d9cb6d97078ae526af84a8eb86f.